common/system/tasks/radio_manager.h

```diff
@@ -5,59 +5,73 @@
 #ifndef COMMON_SYSTEM_TASKS_RADIO_MANAGER_H_
 #define COMMON_SYSTEM_TASKS_RADIO_MANAGER_H_
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <stdint.h>
 
 #include "board_config_common.h"
 #include "circular_buffer.h"
 #include "data_structures.h"
 #include "filters.h"
 #include "hardware_manager.h"
+#include "hm_timer.h"
 #include "radio_packet_types.h"
 
-// Struct to keep track of when messages were last sent
-typedef struct {
-  uint32_t propStuffLastSent;
-  uint32_t orientationLastSent;
-  uint32_t positionLastSent;
-  uint32_t lineCutterLastSent;
-  uint32_t lineCutterVarsLastSent;
-  uint32_t altInfoLastSent;
-  uint32_t hardwareStatusLastSent;
-  uint32_t cliStringLastSent;
-} DataTransmitState_s;
-
 #define RADIO_MAX_CALLBACKS 10
-typedef void (*RadioCallback_t)(RadioRecievedPacket_s*);
+typedef void (*RadioCallback_t)(RadioDecodedRecievedPacket_s*);
+
+struct PacketTimerCollection {
+  HmTickTimer orientationTimer{10};
+  HmTickTimer positionTimer{10};
+  HmTimer altInfoTimer{1213};
+  HmTimer lineCutterDataTimer{1000};
+  HmTimer lineCutterVarsTimer{1000};
+  HmTickTimer pyroContTimer{1};
+};
+
+class RadioManager {
+ public:
+  RadioManager() = default;
+
+  void init(int id);
+  void tick();
+
+  void transmitData(SensorData_s* sensorData, FilterData_s* filterData,
+                    uint8_t state);
+
+  // Send a string over radio
+  // Note that this BLOCKS for up to 150ms!!
+  void transmitString(uint8_t* data, size_t len);
+
+  void addMessageCallback(RadioCallback_t callback);
+
+  static void InitAll();
```

Comment:
See above comment about having a singular radio manager rather than having some static methods that operate on some existing static list of them all.

Reply:
There's a good chunk of state per radio. Maybe we can find a new name for "thing that manages one radio"?

Reply:
Yeah sure, if you want to add another intermediary one-radio-manager-something that's fine with me, and then maybe the radio manager delegates to those when called for a specific one.